packages/core/contracts-ovm/insured-bridge/implementation/L1_BridgeRouter.sol

```diff
@@ -0,0 +1,127 @@
+        DepositState depositState;
+        DepositType depositType;
+        // The following params are set by the L2 depositor:
+        address l1Recipient;
+        address l2Token;
+        uint256 amount;
+        uint256 maxFee;
+        // Params inferred by this contract:
+        address l1Token;
+        // The following params are inferred and set by the L2 deposit contract:
+        address l2Sender;
+        uint256 depositTimestamp;
+        uint256 depositNetId;
+        // Relayer will compute the realized fee considering the amount of liquidity in this contract and the pending
+        // withdrawals at the depositTimestamp.
+        uint256 realizedFee;
+        // A deposit can have both a slow and an instant relayer if a slow relay is "sped up" from slow to instant. In
+        // these cases, we want to store both addresses for separate payouts.
+        address slowRelayer;
+        address instantRelayer;
+        // TODO: Not sure how this will be used or why its stored but its in the interface doc
+        bytes priceRequestAncillaryData;
+    }
+    // Associates each deposit with a unique ID.
+    mapping(uint256 => Deposit) deposits;
+
+    event AddedDepositContract(address indexed l2DepositContract, uint256 indexed netId);
+    event WhitelistToken(address indexed l2Token, address indexed l1Token);
+    event DepositRelayed(
+        address indexed sender,
+        address recipient,
+        address indexed l2Token,
+        address indexed l1Token,
+        address relayer,
+        uint256 amount,
+        uint256 netId,
+        uint256 realizedFee,
+        uint256 maxFee
+    );
+    event RelaySpedUp(uint256 indexed depositId, address indexed fastRelayer, address indexed slowRelayer);
+    event FinalizedRelay(uint256 indexed depositId, address indexed caller);
+    event RelayDisputeSettled(uint256 indexed depositId, address indexed caller, bool disputeSuccessful);
+
+    // TODO: We can't use @openzeppelin/Ownable until we bump this contract to Solidity 0.8
+    address public owner;
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Not owner");
+        _;
+    }
+
+    constructor(
+        address _finder,
+        address _crossDomainMessenger,
+        address _owner,
+        uint256 _optimisticOracleLiveness
+    ) OVM_CrossDomainEnabled(_crossDomainMessenger) {
+        finder = _finder;
+        owner = _owner;
+        optimisticOracleLiveness = _optimisticOracleLiveness;
+    }
+
+    // Admin functions
+
+    function addDepositContract(address depositContract, uint256 netId) public onlyOwner {}
+
+    function whitelistToken(address l1Token, address l2Token) public onlyOwner {}
+
+    function pauseL2Deposits() public onlyOwner {}
+
+    // Liquidity provider functions
+
+    function deposit(address l1Token, uint256 amount) public {}
+
+    function withdraw(address lpToken, uint256 amount) public {}
+
+    // Relayer functions
+
+    function relayDeposit(
```

Comment:
So @chrismaree IIUC, every `l2Token` will be automatically mapped to its `l1Token` via the `whitelistedTokens` mapping. Therefore, the `sender` from L2 needs to be aware of this mapping and set the `l1Token` address correctly in their original L2 deposit?

Reply:
correct. My thinking is the only way to whitelist a token on L2 is by calling `whitelistToken` on this contract which will then call over the canonical bridge the `depositContract` to append to the mapping there. That way the sender does not need to know about that on L2 as it will actually be there in the contract already & the whitelisting process will be atomic over both networks.

Reply:
notice how the function `whitelistToken` on the `L2_bridgeDepositBox` has the modifer `onlyFromCrossDomainAccount(l1Owner)` such that only this contract can whitelist tokens on L2